src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2344,11 +2345,50 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
             $container->removeDefinition(HttpClient::class);
         }
 
-        if ($this->readConfigEnabled('http_client.retry_failed', $container, $retryOptions)) {
+        if ($hasRetryFailed = $this->readConfigEnabled('http_client.retry_failed', $container, $retryOptions)) {
             $this->registerRetryableHttpClient($retryOptions, 'http_client', $container);
         }
 
-        $httpClientId = ($retryOptions['enabled'] ?? false) ? 'http_client.retryable.inner' : ($this->isInitializedConfigEnabled('profiler') ? '.debug.http_client.inner' : 'http_client');
+        if ($hasUriTemplate = $config['uri_template']['enabled']) {
+            if (!class_exists(UriTemplateHttpClient::class)) {
+                throw new LogicException('Support for URI template requires symfony/http-client 6.3 or higher, try upgrading.');
+            }
+
+            $expanderDefinition = $expanderCallable = null;
+            if ($expanderId = $config['uri_template']['expander'] ?? null) {
+                $expanderCallable = new Reference($expanderId);
+            } elseif (class_exists(\GuzzleHttp\UriTemplate\UriTemplate::class)) {
```

Comment:
we should use ContainerBuilder::willBeAvailable for these checks
we're missing a check for league's